File: day-2/part-1.jl
course = readlines("input.txt") .|>
	line -> (dir = line[1], value = parse(Int, split(line)[2]))

submarine = (x = 0, z = 0)

for cmd in course
	if cmd.dir == 'f'
		global submarine = merge(submarine, (x = submarine.x + cmd.value,))
	elseif cmd.dir == 'u'
		global submarine = merge(submarine, (z = submarine.z - cmd.value,))
	elseif cmd.dir == 'd'
		global submarine = merge(submarine, (z = submarine.z + cmd.value,))
	end
end

print(submarine.x * submarine.z)

File: day-2/part-1.exs
File.read!("input.txt")
|> String.split("\n", trim: true)
|> Enum.map(fn l ->
	%{ :dir => String.first(l), :value => l |> String.last |> Integer.parse |> elem(0) }
end)
|> Enum.reduce(%{:x => 0, :z => 0}, fn cmd, acc ->
	case cmd.dir do
		"f" -> %{ acc | :x => acc.x + cmd.value}
		"d" -> %{ acc | :z => acc.z + cmd.value}
		"u" -> %{ acc | :z => acc.z - cmd.value}
		_ -> acc
	end
end)
|> (fn sub -> sub.x * sub.z end).()
|> IO.puts
